activejob/lib/active_job/exceptions.rb

```diff
@@ -122,7 +127,8 @@ def retry_job(options = {})
     end
 
     private
-      def determine_delay(seconds_or_duration_or_algorithm:, executions:, jitter:)
+      def determine_delay(seconds_or_duration_or_algorithm:, executions:, jitter: nil)
+        jitter ||= self.class.default_retry_jitter
```

Comment:
I'd expect passing `jitter: nil` to override the default rather than fall back to it. That suggests requiring the jitter kwarg here and using `jitter: self.class.retry_jitter` in `retry_on` kwargs. Then we'd need to treat `jitter: nil` as 0 here.

Reply:
Started on this but got stung by some time that didn't seem to be wholly frozen or some such (3.45 seconds seemed to be added consistently).

```patch
diff --git a/activejob/lib/active_job/exceptions.rb b/activejob/lib/active_job/exceptions.rb
index 71bb7d615e..46616b38dd 100644
--- a/activejob/lib/active_job/exceptions.rb
+++ b/activejob/lib/active_job/exceptions.rb
@@ -126,8 +126,11 @@ def retry_job(options = {})
     end
 
     private
+      JITTER_DEFAULT = Object.new
+
       def determine_delay(seconds_or_duration_or_algorithm:, executions:, jitter: nil)
-        jitter ||= self.class.retry_jitter
+        jitter = jitter == JITTER_DEFAULT ? self.class.retry_jitter : jitter.to_i
+
         case seconds_or_duration_or_algorithm
         when :exponentially_longer
           ((executions**4) + (Kernel.rand((executions**4) * jitter))) + 2
diff --git a/activejob/test/cases/exceptions_test.rb b/activejob/test/cases/exceptions_test.rb
index 0304c56124..6d2487cf39 100644
--- a/activejob/test/cases/exceptions_test.rb
+++ b/activejob/test/cases/exceptions_test.rb
@@ -154,6 +154,26 @@ class ExceptionsTest < ActiveSupport::TestCase
     ActiveJob::Base.retry_jitter = old_jitter
   end
 
+  test "disable retry jitter" do
+    travel_to Time.now
+
+    Kernel.stub(:rand, ->(arg) { arg }) do
+      RetryJob.perform_later "DisabledJitterError", 5, :log_scheduled_at
+
+      assert_equal [
+        "Raised DisabledJitterError for the 1st time",
+        "Next execution scheduled at #{Time.now.to_f}",
+        "Raised DisabledJitterError for the 2nd time",
+        "Next execution scheduled at #{Time.now.to_f}",
+        "Raised DisabledJitterError for the 3rd time",
+        "Next execution scheduled at #{Time.now.to_f}",
+        "Raised DisabledJitterError for the 4th time",
+        "Next execution scheduled at #{Time.now.to_f}",
+        "Successfully completed job"
+      ], JobBuffer.values
+    end
+  end
+
   test "custom wait retrying job" do
     travel_to Time.now
 
diff --git a/activejob/test/jobs/retry_job.rb b/activejob/test/jobs/retry_job.rb
index 112d672006..52107110b7 100644
--- a/activejob/test/jobs/retry_job.rb
+++ b/activejob/test/jobs/retry_job.rb
@@ -4,6 +4,7 @@
 require "active_support/core_ext/integer/inflections"
 
 class DefaultsError < StandardError; end
+class DisabledJitterError < StandardError; end
 class FirstRetryableErrorOfTwo < StandardError; end
 class SecondRetryableErrorOfTwo < StandardError; end
 class LongWaitError < StandardError; end
@@ -18,6 +19,7 @@ class CustomDiscardableError < StandardError; end
 
 class RetryJob < ActiveJob::Base
   retry_on DefaultsError
+  retry_on DisabledJitterError, jitter: nil
   retry_on FirstRetryableErrorOfTwo, SecondRetryableErrorOfTwo, attempts: 4
   retry_on LongWaitError, wait: 1.hour, attempts: 10
   retry_on ShortWaitTenAttemptsError, wait: 1.second, attempts: 10
```

Reply:
@jeremy / @kaspth, I agree that nil being treated as zero makes sense. I’d still like to see the kwarg optional if possible, unless there’s a sound reason not to. It’s not super critical but does make dual booting Rails 6.0 and 6.1 a _little_ easier at Github.

Reply:
> see the kwarg optional if possible

That's the purpose behind `JITTER_DEFAULT = Object.new` and comparing with that. That lets us detect `retry_job jitter: nil` while also supporting `retry_job` and inferring jitter from `self.class.retry_jitter`.